x-pack/plugins/event_log/server/es/cluster_client_adapter.ts

```diff
@@ -188,6 +194,128 @@ export class ClusterClientAdapter<TDoc extends { body: AliasAny; index: string }
+    try {
+      const esClient = await this.elasticsearchClientPromise;
+      const { body: templates } = await esClient.indices.getTemplate(
+        { name: indexTemplatePattern },
+        { ignore: [404] }
+      );
+      return templates;
+    } catch (err) {
+      throw new Error(`error getting existing legacy index templates: ${err.message}`);
+    }
+  }
+
+  public async setLegacyIndexTemplateToHidden(
+    indexTemplateName: string,
+    currentIndexTemplate: estypes.IndicesTemplateMapping
+  ): Promise<void> {
+    try {
+      const esClient = await this.elasticsearchClientPromise;
+      await esClient.indices.putTemplate({
+        name: indexTemplateName,
+        body: {
+          ...currentIndexTemplate,
+          settings: {
+            ...currentIndexTemplate.settings,
+            'index.hidden': true,
+          },
+        },
+      });
+    } catch (err) {
+      throw new Error(
+        `error setting existing legacy index template ${indexTemplateName} to hidden: ${err.message}`
+      );
+    }
+  }
+
+  public async getExistingIndices(
+    indexPattern: string
+  ): Promise<estypes.IndicesGetSettingsResponse> {
+    try {
+      const esClient = await this.elasticsearchClientPromise;
+      const { body: indexSettings } = await esClient.indices.getSettings(
+        { index: indexPattern },
+        { ignore: [404] }
+      );
+      return indexSettings;
+    } catch (err) {
+      throw new Error(
+        `error getting existing indices matching pattern ${indexPattern}: ${err.message}`
+      );
+    }
+  }
+
+  public async setIndexToHidden(indexName: string): Promise<void> {
+    try {
+      const esClient = await this.elasticsearchClientPromise;
+      await esClient.indices.putSettings({
+        index: indexName,
+        body: {
+          settings: {
+            'index.hidden': true,
+          },
+        },
+      });
+    } catch (err) {
+      throw new Error(`error setting existing index ${indexName} to hidden: ${err.message}`);
+    }
+  }
+
+  public async getExistingIndexAliases(
+    indexPattern: string
+  ): Promise<estypes.IndicesGetAliasResponse> {
+    try {
+      const esClient = await this.elasticsearchClientPromise;
+      const { body: indexAliases } = await esClient.indices.getAlias(
+        { index: indexPattern },
+        { ignore: [404] }
+      );
+      return indexAliases;
+    } catch (err) {
+      throw new Error(
+        `error getting existing index aliases matching pattern ${indexPattern}: ${err.message}`
+      );
+    }
+  }
+
+  public async setIndexAliasToHidden(
+    indexName: string,
+    currentAliases: estypes.IndicesGetAliasIndexAliases
+  ): Promise<void> {
+    try {
+      const esClient = await this.elasticsearchClientPromise;
+      await esClient.indices.updateAliases({
+        body: {
+          actions: Object.keys(currentAliases.aliases).map((aliasName) => {
+            const existingAliasOptions = pick(currentAliases.aliases[aliasName], [
```

Comment:
Are there some fields you can't PUT back up, or why are we doing a `pick` here?

Reply:
Yes, this API supports multiple `action` types and only some of these fields are applicable to the `add` action. This is to ensure we're only sending the appropriate fields for the `add` action. It's a little clunky :-/